compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -564,10 +564,7 @@ fn link_staticlib<'a>(
                 .extract_bundled_libs(path, tempdir.as_ref(), &relevant_libs)
                 .unwrap_or_else(|e| sess.dcx().emit_fatal(e));
 
-            // We sort the libraries below
-            #[allow(rustc::potential_query_instability)]
-            let mut relevant_libs: Vec<Symbol> = relevant_libs.into_iter().collect();
-            relevant_libs.sort_unstable();
+            let relevant_libs: Vec<Symbol> = relevant_libs.into_iter().collect();
```

Comment:
That sorting actually looks like a bug because sorting by `Symbol` is not stable. And I agree that at least the second `collect` is not necessary since we are deduplicating via an `FxIndexMap`.